File: typescript/tiktok/video-views.ts
// How do I get the view count of a TikTok video?
// https://openhandle.dev/questions/how-do-i-get-the-view-count-of-a-tiktok-video
// Run: node tiktok/video-views.ts (Node 24 or newer)

import { OpenHandle } from '@openhandle/sdk';

const openhandle = new OpenHandle({ apiKey: process.env.OPENHANDLE_TEST_KEY! });

const { data, capturedAt } = await openhandle.tiktok.post('920100000000000001').get({ freshness: 'live' });

console.log(data.metrics.views, data.metrics.likes, capturedAt); // 23000 1200 ...

File: python/tiktok/video_views.py
# How do I get the view count of a TikTok video?
# https://openhandle.dev/questions/how-do-i-get-the-view-count-of-a-tiktok-video
# Run: python tiktok/video_views.py

import os

from openhandle import OpenHandle

openhandle = OpenHandle(api_key=os.environ["OPENHANDLE_TEST_KEY"])

response = openhandle.tiktok.post("920100000000000001").get(freshness="live")

print(response.data["metrics"]["views"], response.data["metrics"]["likes"], response.captured_at)  # 23000 1200 ...
